### PR TITLE
Update CSI sanity to v4.2.0

### PR DESCRIPTION
**Pull request by chrishenzie**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR updates the CSI sanity library to the most recent release, v4.2.0.

This is important for asserting correct driver behavior and is a prerequisite for additional tests we will want to run against this driver, namely for the [new NodePublishVolume behavior tied to the new access modes](https://github.com/container-storage-interface/spec/blob/v1.5.0/spec.md#nodepublishvolume).

**Special notes for your reviewer**:
Please see #818 for more details on the skipped tests and possible solutions.

**Does this PR introduce a user-facing change?**:
```release-note
Updates the CSI sanity test suite to v4.2.0
```

/assign @msau42 
/assign @mattcary

### CODE REVIEW

File: pkg/gce-cloud-provider/compute/fake-gce.go [hunk 1]
```diff
@@ -293,9 +293,6 @@ func (cloud *FakeCloudProvider) InsertDisk(ctx context.Context, project string,
 }
 
 func (cloud *FakeCloudProvider) DeleteDisk(ctx context.Context, project string, volKey *meta.Key) error {
-	if _, ok := cloud.disks[volKey.Name]; !ok {
-		return notFoundError()
-	}
```
Comment (chrishenzie) [inline pkg/gce-cloud-provider/compute/fake-gce.go:298]: For context, this was removed to align behavior with the real implementation.

**Comment by chrishenzie** [comment]
This is in a reviewable state now, PTAL when you have a moment

File: go.sum [hunk 2]
```diff
@@ -607,6 +612,7 @@ github.com/prometheus/procfs v0.6.0/go.mod h1:cz+aTbrPOrUb4q7XlbU9ygM+/jj0fzG6c1
 github.com/prometheus/tsdb v0.7.1/go.mod h1:qhTCs0VvXwvX/y3TZrWD7rabWM+ijKTux40TwIPHuXU=
 github.com/quobyte/api v0.1.8/go.mod h1:jL7lIHrmqQ7yh05OJ+eEEdHr0u/kmT1Ff9iHd+4H6VI=
 github.com/remyoudompheng/bigfft v0.0.0-20170806203942-52369c62f446/go.mod h1:uYEyJGbgTkfkS4+E/PavXkNJcbFIpEtjt2B0KDQ5+9M=
+github.com/robertkrimen/otto v0.0.0-20200922221731-ef014fd054ac/go.mod h1:xvqspoSXJTIpemEonrMDFq6XzwHYYgToXWj5eRX1OtY=
```
Comment (mattcary) [inline go.sum:615]: Did you see who pulled this in?

I guess it's a transitive dependency somewhere, but it wasn't obvious from the vendor changes below. I guess adding a random javascript parsing library is just standard modern programming process :-/

Comment (chrishenzie) [reply go.sum:615]: From what I recall, otto is used in K8s to script CSI behavior in e2e tests, maybe that's where it came from

Comment (mattcary) [reply go.sum:615]: Fair enough, thanks.

/lgtm
/approve

**Pull request merged**